### PR TITLE
Add common parent class for application classes

### PR DESCRIPTION
**Pull request by irfano**
We need different application classes for UI test and real app to use Hilt. The main reason for this requirement is that we can't use `@Inject` annotation in the test application because the test application handles injection differently in Hilt.

**[WordPress](https://github.com/wordpress-mobile/WordPress-Android/blob/7f427fe69e0cc980b04b8cfd6dfb5d8e98820c0a/WordPress/src/main/java/org/wordpress/android/WordPress.kt):** Abstract superclass application which contains common jobs for real and test applications.
**[WordPressApp](https://github.com/wordpress-mobile/WordPress-Android/blob/7f427fe69e0cc980b04b8cfd6dfb5d8e98820c0a/WordPress/src/main/java/org/wordpress/android/WordPressApp.java):** Real application
**[WordPressTest](https://github.com/wordpress-mobile/WordPress-Android/blob/7f427fe69e0cc980b04b8cfd6dfb5d8e98820c0a/WordPress/src/androidTest/java/org/wordpress/android/WordPressTest.kt):** UI test application. The only difference from `WordPressApp` will be handling injection differently. Difference will be in next PRs.

We need the ability to initialize Hilt whenever we want without `@Inject` annotation. To provide this, I added `WordPress`, which contains common stuff for `WordPressTest` and `WordPressApp`.

This PR can seem meaningless before future PRs, but it will be useful after adding Hilt.

To test:
Build and run the app. It should be running without an error.

## Regression Notes
1. Potential unintended areas of impact
Building and tests might be broken.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I smoke tested the app, and it's working as expected. Also, tests are working fine.

3. What automated tests I added (or what prevented me from doing so)
No tests are added since no new feature is added.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by irfano** [comment]
@ravishanker, I need to update `WordPressTest`'s superclass with `BaseWordPress` to continue this PR. But after doing that, tests won't work because `(WordPress)` casts will fail in tests. There are 225 `WordPress` casting in the code. So, I need to change all these castings to cast `BaseWordPress,` but it will change a lot of files.

Alternatively, I can use `WordPress` as a superclass, as I did in https://github.com/wordpress-mobile/WordPress-Android/pull/15653. Then I can add another class for real applications. But the important **drawback** is that I will have to change the application's name in this way, like `WordPress` -> `WordPressProd.`
https://github.com/wordpress-mobile/WordPress-Android/blob/345702735819d91adf32c9f64e8f01ec91973809/WordPress/src/main/AndroidManifest.xml#L56
I think application name change is a critical change. It could be used for deeplinks or in some external services.

I couldn't find a better solution than changing 225 casting lines from `(WordPress)` to `(BaseWordPress)`. Do you think I should go this way?

**Comment by ravishanker** [comment]
> `WordPress`

@irfano  -  It is never easy these kind of structural changes.  Are these casts only in tests or in the application code?  If it is in tests I wouldn't worry about it.  In any case, it would be good to socialize this change with the rest of android community.  Please post about it to #android-dev slack channel and get some feedback on the proposed change.

**Comment by irfano** [comment]
> Are these casts only in tests or in the application code?

No, it's in the application code. Thanks, I'll share this to get feedback.

**Comment by irfano** [comment]
After receiving feedback from Slack, I decided to name superclass as `WordPress` to avoid causing changes on 255 files. We couldn't come up with any drawbacks of changing the application class name from `WordPress` to `WordPressApp`.

File: WordPress/src/androidTest/java/org/wordpress/android/WordPressTest.kt [hunk 1]
```diff
@@ -2,9 +2,9 @@ package org.wordpress.android
 
 import org.wordpress.android.modules.DaggerAppComponentTest
 
-class WordPressTest : WordPress() {
+class WordPressTest : WordPressApp() {
```
Comment (irfano) [inline WordPress/src/androidTest/java/org/wordpress/android/WordPressTest.kt:5]: For now, `WordPressTest` is extended from `WordPressApp`. When Hilt added, `WordPressTest` will extend `WordPress` and inject and init `AppInitializer` differently from `WordPressApp`.

Comment (ravishanker) [reply WordPress/src/androidTest/java/org/wordpress/android/WordPressTest.kt:5]: Sounds good.  I've been able to build and run without error

**Review by ravishanker — APPROVED** [review]
LGTM 👍

**Pull request merged**